FhirToDataLake/Synapse-search.yml

```diff
@@ -0,0 +1,210 @@
+
+      Write-Host 'Assign Storage Blob Data Contributor for Synapse Workspace to Storage.'
+      az role assignment create --role "Storage Blob Data Contributor" --assignee-object-id $synapsePrincipalId --scope $storageScope --assignee-principal-type ServicePrincipal
+      
+      Write-Host 'Assign Storage Blob Data Contributor for service connection account to Storage.'
+      az role assignment create --role "Storage Blob Data Contributor" --assignee $serviceConnectionPrincipalId --scope $storageScope
+
+      Write-Host 'Assign Synapse SQL Administrator for service connection account to Synapse SQL pool.'
+      az synapse role assignment create --workspace-name $(synapseWorkspaceName) --role "Synapse SQL Administrator" --assignee $serviceConnectionPrincipalId
+      
+      # Wait 1 min in case the assignments don't enable.
+      Start-Sleep -s 60
+
+- task: AzurePowerShell@5
+  displayName: Set up EXTERNAL TABLEs and VIEWs on Synapse.
+  inputs: 
+    azureSubscription: 'ResoluteOpenSource'
+    azurePowerShellVersion: latestVersion
+    ScriptType: inlineScript
+    Inline: |
+      Install-Module -Name Az.Synapse -RequiredVersion 0.7.0 -Force
+      Import-Module Az.Synapse
+      Copy-Item "$(System.DefaultWorkingDirectory)\FhirToDataLake\scripts\.readme.txt" -Destination "."
+
+      # Retry to avoid the warming up of Synapse SQL pool and other potential issues
+      # see https://docs.microsoft.com/en-us/sql/relational-databases/errors-events/mssqlserver-42109-database-engine-error?view=sql-server-ver15
+
+      $attempts = 1
+      $maxAttempts = 3
+      do
+      {
+          try
+          {
+              . "$(System.DefaultWorkingDirectory)\FhirToDataLake\scripts\Set-SynapseEnvironment.ps1"  -SynapseWorkspaceName $(synapseWorkspaceName) -StorageName $(storageAccountName) -Database $(databaseName) -SqlScriptCollectionPath $(System.DefaultWorkingDirectory)\FhirToDataLake\scripts\sql\Resources
+              break;
+          }
+          catch [Exception]
+          {
+              Write-Host $_.Exception.Message
+          }
+
+          # exponential backoff delay
+          $attempts++
+          if ($attempts -le $maxAttempts) {
+              $retryDelaySeconds = [math]::Pow(2, $attempts)
+              $retryDelaySeconds = $retryDelaySeconds - 1  # Exponential Backoff Max == (2^n)-1
+              Write-Host("Set up EXTERNAL TABLEs and VIEWs failed. Waiting " + $retryDelaySeconds + " seconds before attempt " + $attempts + " of " + $maxAttempts + ".")
+              Start-Sleep $retryDelaySeconds 
+          }
+          else {
+              Write-Error $_.Exception.Message
+          }
+      } while ($attempts -le $maxAttempts)
+
+- script: python -m pip install requests pyodbc pandas
+  displayName: Install requirements for validate script
+
+- task: PythonScript@0
+  displayName: Validate results
```

Comment:
Data Validation?

Reply:
Update to Validate data, just to keep consistence with a verb in the beginning